homeassistant/components/ezviz/select.py

```diff
@@ -21,19 +27,105 @@
 PARALLEL_UPDATES = 1
 
 
+class EzvizSelectEntityActionBase:
+    """Base class to handle retrieval and selection of a EzvizSelectEntity."""
+
+    current_option: Callable[[EzvizSelect], str | None]
+    select_option: Callable[[EzvizSelect, str, str], None]
+
+
+class AlarmSoundModeAction(EzvizSelectEntityActionBase):
+    """Class dedicated to Alarm Sound Mode."""
+
+    @staticmethod
+    def current_option(ezvizSelect: EzvizSelect) -> str | None:
+        """Return the selected entity option to represent the entity state."""
+        sound_mode_value = getattr(
+            SoundMode, ezvizSelect.data[ezvizSelect.entity_description.key]
+        ).value
+        if sound_mode_value in [0, 1, 2]:
+            return ezvizSelect.options[sound_mode_value]
+
+        return None
+
+    @staticmethod
+    def select_option(ezvizSelect: EzvizSelect, serial: str, option: str) -> None:
+        """Change the selected option."""
+        sound_mode_value = ezvizSelect.options.index(option)
+
+        try:
+            ezvizSelect.coordinator.ezviz_client.alarm_sound(
+                serial, sound_mode_value, 1
+            )
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Cannot set Warning sound level for {ezvizSelect.entity_id}"
+            ) from err
+
+
+class BatteryWorkModeAction(EzvizSelectEntityActionBase):
+    """Class dedicated to Battery Work Mode."""
+
+    @staticmethod
+    def current_option(ezvizSelect: EzvizSelect) -> str | None:
+        """Return the selected entity option to represent the entity state."""
+        battery_work_mode = getattr(
+            BatteryCameraWorkMode,
+            ezvizSelect.data[ezvizSelect.entity_description.key],
+            BatteryCameraWorkMode.UNKNOWN,
+        )
+        if battery_work_mode == BatteryCameraWorkMode.UNKNOWN:
+            return None
+
+        return battery_work_mode.name.lower()
+
+    @staticmethod
+    def select_option(ezvizSelect: EzvizSelect, serial: str, option: str) -> None:
+        """Change the selected option."""
+        battery_work_mode = getattr(BatteryCameraWorkMode, option.upper())
+        try:
+            ezvizSelect.coordinator.ezviz_client.set_battery_camera_work_mode(
+                serial, battery_work_mode.value
+            )
+
+        except (HTTPError, PyEzvizError) as err:
+            raise HomeAssistantError(
+                f"Cannot set battery work mode for {ezvizSelect.entity_id}"
+            ) from err
+
+
 @dataclass(frozen=True, kw_only=True)
 class EzvizSelectEntityDescription(SelectEntityDescription):
     """Describe a EZVIZ Select entity."""
 
     supported_switch: int
+    action_handler: type[EzvizSelectEntityActionBase]
```

Comment:
```suggestion
    current_option_fn: Callable[[EzvizSelect], str | None]
    select_option_fn: Callable[[EzvizSelect, str, str], None]
```

Reply:
Or maybe even just replace `EzvizSelect` with `ezvizSelect.data[ezvizSelect.entity_description.key]` in the `current_option_fn` and the `ezviz_client` in the `select_option_fn`